### PR TITLE
[6] feat(1270): add hostname for build

### PR DESCRIPTION
**Pull request by d2lam**
Similar to https://github.com/screwdriver-cd/executor-k8s/pull/100
Related: https://github.com/screwdriver-cd/screwdriver/issues/1270

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -317,16 +328,26 @@ class K8sVMExecutor extends Executor {
                         `${JSON.stringify(resp.body.status, null, 2)}`);
                 }
 
+                const updateConfig = {
+                    apiUri: this.ecosystem.api,
+                    buildId: config.buildId,
```
Comment (tkyi) [inline index.js]: don't need `config.` anymore

File: index.js [hunk 2]
```diff
@@ -317,16 +327,23 @@ class K8sVMExecutor extends Executor {
...
 
+                const updateConfig = {
+                    apiUri: this.ecosystem.api,
+                    buildId,
+                    token
+                };
+
+                if (resp.body.spec && resp.body.spec.nodeName) {
+                    updateConfig.stats = {
+                        hostname: resp.body.spec.nodeName
```
Comment (minz1027) [inline index.js]: if build's stats has `queueEnterTime` in it, this will override that.

Comment (d2lam) [reply index.js]: It will handle the merging logic inside the API
this won't merge or override

Comment (minz1027) [reply index.js]: 👍

**Pull request merged**